src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -207,6 +207,12 @@
     MultilineConcatenateDelimiterGrammar=AnyNumberOf(
         Ref("MultilineConcatenateNewline"), min_times=1, allow_gaps=False
     ),
+    QuotedParameterNameSegment=RegexParser(
+        r'"[^"]+"', CodeSegment, name="quoted_parameter", type="parameter"
+    ),
+    ParameterNameGrammar=OneOf(
+        Ref("ParameterNameSegment"), Ref("QuotedParameterNameSegment")
+    ),
```

Comment:
This is the core of the change.  The other changes in this file are mechanical replacements of `ParameterNameSegment` with `ParameterNameGrammar`.

Reply:
Hmmm.... I'm starting to have second thoughts about my suggestion here.

I thought when I was first reviewing this, that `ParameterNameSegment` was a postgres dialect specific thing, but it's used in ansi dialect - and lots of others. And Postgres still inherits some of the syntax from ansi version.

I also noticed BigQuery already has this set as a `OneOf` of normal and quoted.

So think we should either reverse your last change. Or alternatively change it in all the other dialects - but that's a much bigger change.

Sorry about that!

Reply:
No worries.  I'll revert the changes and see where we land.

Reply:
Get a chance to look at this @benji-york ? Can we just revert last few changes or is it not as simple as that?

And apologies again for sending you on this wild goose chase!

Reply:
My git-fu has failed me and I've made a royal mess of this branch.  I won't have time to work on this for at least the next few days.  If you want some movement on this before I'm able to get to it, feel free to lift whatever you want from this work and merge it.

One consideration: just about everything is quotable in Postgres, so I'm sure there is more that should be done along this route.  On the other hand, we shouldn't let the perfect be the enemy of the good.